+bids/layout.m

```diff
@@ -182,159 +176,39 @@
 
     file_list = return_file_list(modality, subject, schema);
 
-    for i = 1:numel(file_list)
-
-      subject = bids.internal.append_to_layout(file_list{i}, subject, modality, schema);
-
-    end
-
-  end
-
-end
-
-function subject = parse_dwi(subject, schema)
+    for iFile = 1:size(file_list, 1)
 
-  % --------------------------------------------------------------------------
-  %  Diffusion imaging data
-  % --------------------------------------------------------------------------
+      [subject, parsing] = bids.internal.append_to_layout(file_list{iFile}, subject, modality, schema);
 
-  modality = 'dwi';
-  pth = fullfile(subject.path, modality);
+      if ~isempty(parsing)
 
-  if exist(pth, 'dir')
+        subject = index_dependencies(subject, modality, file_list{iFile});
 
-    subject = bids.internal.add_missing_field(subject, modality);
+        switch subject.(modality)(end).suffix
 
-    file_list = return_file_list(modality, subject, schema);
+          case 'asl'
 
-    for i = 1:numel(file_list)
+            subject.(modality)(end).meta = [];
 
-      subject = bids.internal.append_to_layout(file_list{i}, subject, modality, schema);
+            subject.(modality)(end).meta = bids.internal.get_metadata(subject.(modality)(iFile).metafile);
 
-      % if this file is a nifti image we add the bval and bvec as dependencies
-      if ~isempty(subject.(modality)) && ...
-              any(strcmp(subject.(modality)(end).ext, {'.nii', '.nii.gz'}))
+            aslcontext_file = strrep(subject.perf(end).filename, ...
+                                     ['_asl' subject.perf(end).ext], ...
+                                     '_aslcontext.tsv');
+            subject.(modality)(end).dependencies.context = manage_tsv( ...
```

Comment:
Not sure that asl and aslcontext means, but if it only replaces suffix, will not the `_aslcontext.tsv` appear in group dependencies?

Reply:
it will but it is also needed by `layout` to take care of the M0 so at the moment, it might be better to leave it there.